pkg/cmd/variable/delete/delete.go

```diff
@@ -0,0 +1,138 @@
+package delete
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/variable/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	Config     func() (config.Config, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	VariableName  string
+	OrgName       string
+	EnvName       string
+	UserVariables bool
+	Application   string
+}
+
+func NewCmdDelete(f *cmdutil.Factory, runF func(*DeleteOptions) error) *cobra.Command {
+	opts := &DeleteOptions{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "delete <variable-name>",
+		Short: "Delete variables",
+		Long: heredoc.Doc(`
+			Delete a variable on one of the following levels:
+			- repository (default): available to Actions runs or Dependabot in a repository
+			- environment: available to Actions runs for a deployment environment in a repository
+			- organization: available to Actions runs, Dependabot, or Codespaces within an organization
+			- user: available to Codespaces for your user
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--org`, `--env`, or `--user`", opts.OrgName != "", opts.EnvName != "", opts.UserVariables); err != nil {
+				return err
+			}
+
+			opts.VariableName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return removeRun(opts)
+		},
+		Aliases: []string{
+			"remove",
+		},
+	}
+	cmd.Flags().StringVarP(&opts.OrgName, "org", "o", "", "Delete a variable for an organization")
+	cmd.Flags().StringVarP(&opts.EnvName, "env", "e", "", "Delete a variable for an environment")
+	cmd.Flags().BoolVarP(&opts.UserVariables, "user", "u", false, "Delete a variable for your user")
```

Comment:
No necessary for variables. Lets remove this and other code related to user level variables.